### DISCUSSION TITLE
Page protected by authentication - should or shouldn't be cached

### DISCUSSION THREAD

**Discussion by rsvoboda** [quarkusio/quarkus#45187] [Design Discussions]
Page protected by authentication - should or shouldn't be cached (HTTP cache)?

This is followup of discussion on https://github.com/quarkusio/quarkus-quickstarts/pull/1480 where @jedla97 analyzed the root cause of the behavior @gsmet pointed to (https://github.com/quarkusio/quarkus-quickstarts/pull/1480#discussion_r1871768071).

@michalvavrik commented about some ideas (https://github.com/quarkusio/quarkus-quickstarts/pull/1480#discussion_r1885802871) and asked to create an issue to have discussion.

CC @sberyozkin @cescoffier

**Comment by cescoffier** [comment]
What kind of cache are we talking about? Client side cache (browser) or cache annotations.

For the later, except if the cache is able to distinguish users and so handle a different cached value per user, it should not be cached.

For the former, it's tricky. In general, I would say it should not be cached, to avoid potential leak.

**Reply by gsmet** [reply]
I was talking about HTTP cache. What you do with annotations in your code base is your problem :).

**Comment by sberyozkin** [comment]
Headers like Cache-Control and Pragma can be used to instruct proxies, but we don't know what kind of deployment ecosystem is in place. If we set it by default, users in intranet can loose the benefit of caching.  It is also unclear how much caching actually happens over HTTPS which is what authenticated users will work over with. 

I suggest to add a doc section advising users that proxies may have the content cached and if the pages are returned to external authenticated clients, then these cache control headers should be set (it is possible to do it with the configuration or with JAX-RS responses). Additionally, we may advise using MTLS.
